File: src/main/kotlin/org/gameflow/entity/EntityGroupsImpl.kt
package org.gameflow.entity

import java.util.HashMap
import org.gameflow.pass.Pass
import org.gameflow.Game

/**
 * 
 */
public class EntityGroupsImpl: EntityGroups {
    private val entityGroups = HashMap<String, EntityGroup>()

    public override fun groups() = entityGroups.values()

    public override fun createGroup(name: String) : EntityGroup {
        if (entityGroups.containsKey(name)) throw Error("An EnityGroup named '$name' already exists!")
        val group = EntityGroup(name)
        entityGroups.put(name, group)
        return group
    }

    public override fun getGroup(name: String) : EntityGroup {
        val group = entityGroups.get(name)
        if (group == null) throw Error("No EntityGroup named '$name' found!")
        else return group
    }

    public override fun containsGroup(name: String): Boolean = entityGroups.containsKey(name)


    override fun doPassOnChildren(pass : Pass, game : Game) {
        for (group in entityGroups.values()) {
            group.doPass(pass, game)
        }
    }
}

File: src/main/scala/org/gameflow/core/entity/EntityGroupsImpl.scala
package org.gameflow.core.entity

import java.util.HashMap
import org.gameflow.core.pass.Pass
import scala.collection.JavaConversions._
import org.gameflow.core.{Manager, Game}

/**
 *
 */
class EntityGroupsImpl extends EntityGroups {
  private val entityGroups = new HashMap[Symbol, EntityGroup]()

  override def groups(): java.util.Collection[EntityGroup] = entityGroups.values()

  override def createGroup(name: Symbol): EntityGroup = {
    if (entityGroups.containsKey(name)) throw new Error("An EnityGroup named '" + name.name + "' already exists!")
    val group = new EntityGroup(name)
    entityGroups.put(name, group)
    group
  }

  override def getGroup(name: Symbol): EntityGroup = {
    val group = entityGroups.get(name)
    if (group == null) throw new Error("No EntityGroup named '" + name.name + "' found!")
    else group
  }

  override def containsGroup(name: Symbol): Boolean = entityGroups.containsKey(name)


  override def doPassOnChildren(pass: Pass, game: Game) {
    entityGroups.values.foreach{_.doPass(pass, game)}
  }
}